qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/indexing/QuteTemplateIndex.java

```diff
@@ -0,0 +1,132 @@
+package com.redhat.qute.indexing;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.eclipse.lsp4j.Position;
+
+import com.redhat.qute.parser.scanner.Scanner;
+import com.redhat.qute.parser.template.SectionKind;
+import com.redhat.qute.parser.template.scanner.ScannerState;
+import com.redhat.qute.parser.template.scanner.TemplateScanner;
+import com.redhat.qute.parser.template.scanner.TokenType;
+
+public class QuteTemplateIndex {
+
+	private final static List<String> nativeTags = Arrays.asList("each", "for", "if", "else", "set", "let", "elseif");
+
+	private final String templateId;
+
+	private final Path path;
+	private List<QuteIndex> indexes;
+	private FilePositionMap filePositionMap;
+
+	private boolean dirty;
+
+	public QuteTemplateIndex(Path path, String templateId) {
+		this.path = path;
+		this.templateId = templateId;
+		this.indexes = Collections.emptyList();
+	}
+
+	public void collect() throws IOException {
+		this.filePositionMap = null;
+		indexes = new ArrayList<>();
+		String template = Files.readString(path);

```

Comment:
`Files.readString(path)` doesn't exist in Java 1.8. It was introduced in Java 11

Reply:
Qute LS requires Java 11 because it uses the real Qute parser  to parse and collect some errors. And Qute parser requires Java 11.